nsv13/game/objects/pool/components/swimming_felinid.dm

```diff
@@ -0,0 +1,21 @@
+/datum/component/swimming/felinid/enter_pool()
+	var/mob/living/L = parent
+	L.emote("scream")
+	to_chat(parent, "<span class='userdanger'>You get covered in water and start panicing!</span>")
+
+/datum/component/swimming/felinid/process()
+	..()
+	var/mob/living/L = parent
+	if(prob(8))
+		to_chat(parent, "<span class='userdanger'>You can't touch the bottom!</span>")
+		L.emote("scream")
+	else if(prob(16))
```

Comment:
Use a switch for these it is less expensive.

Reply:
nah that's fine, there's practically no performance gain at all

Reply:
I'll do it anyway since it looks nicer

Reply:
Switching a prob() sounds cursed as hell. You'd have to go for a rand() I reckon

Reply:
yea, that's what I did